test-case-reporting/src/test_result_record.ts

```diff
@@ -172,17 +178,24 @@ export class TestResultRecord {
     const buildId = await this.insertTravisBuild(build);
     const jobId = await this.insertTravisJob(job, buildId);
 
-    const formattedResults = results.browsers
-      .map(({results}) => results)
-      .reduce((flattenedArray, array) => flattenedArray.concat(array), [])
-      .map(result => {
-        const testCaseName = this.testCaseName(result);
-        return {
-          ...result,
-          testCaseId: md5(testCaseName),
-          testCaseName,
-        };
-      });
+    let resultList: Array<KarmaReporter.TestResult>;
+
+    if (job.testSuiteType === 'e2e') {
```

Comment:
We could add a field to reports describing the structure of the reporter (mocha vs karma) -- I think that would be clearer than the (currently implicit) connection between e2e tests and the mocha reporter.

Reply:
I agree. We could then make this if(e2e)-elseif(unit|integration)-else(error-unknown)

Reply:
Oh, I mean changing reports from this:
```js
{
  job: // job info
  build: // build info
  results: // results info in some shape
}
```
to this:
```js
{
  job: // job info
  build: // build info
  results: // results info in some shape
  reportType: // mocha or karma
}
```
It's redundant but it's more readable imo, thoughts?